zellij-server/src/pty.rs

```diff
@@ -54,8 +63,9 @@ impl From<&PtyInstruction> for PtyContext {
 }
 
 pub(crate) struct Pty {
+    pub active_pane: Option<PaneId>,
```

Comment:
I'm not too wild about this solution. I think essentially duplicating the state between the `Pty` and the `Screen`/`Tab` is a little error prone. One of those things the developer needs to know about.

That being said, given our current architecture of going through the Pty first before going to the Screen when spawning terminals - I can't think of a trivial way around it. If you have an idea, that would be great. Otherwise we can keep this (I want to say "for now", but if I'm being honest I don't see us changing this ever :) ).

Reply:
The only thing I can think of (without a large refactor) is adding a screen action with an out parameter via a channel. Is that a different kinda of ugly though? Sorry I'm short on ideas here...

Reply:
I think the "right" thing to do would be to change the architecture, or at least formalize some of these paths with a non-blocking shared state somehow... but I feel the right (no-quotation-marks) thing to do for now is to keep your original solution. :)